lib/northbound_hiredis.c

```diff
@@ -0,0 +1,1421 @@
+            sr_val->type = HI_CONTAINER_PRESENCE_T;
+        } else {
+            sr_val->type = HI_CONTAINER_T;
+        }
+        break;
+    case LYS_LIST:
+        sr_val->type = HI_LIST_T;
+        break;
+    case LYS_NOTIF:
+        sr_val->type = HI_NOTIFICATION_T;
+        break;
+    case LYS_ANYXML:
+    case LYS_ANYDATA:
+        any = (struct lyd_node_any *)node;
+        ptr = NULL;
+
+        switch (any->value_type) {
+        case LYD_ANYDATA_STRING:
+        case LYD_ANYDATA_XML:
+        case LYD_ANYDATA_JSON:
+            if (any->value.str) {
+                ptr = strdup(any->value.str);
+                // SR_CHECK_MEM_RET(!ptr, err_info);
+            }
+            break;
+        case LYD_ANYDATA_LYB:
+            /* try to convert into a data tree */
+            if (lyd_parse_data_mem(LYD_CTX(node), any->value.mem, LYD_LYB, LYD_PARSE_STRICT, 0, &tree)) {
+
+				zlog_debug("Failed to convert LYB anyxml/anydata into XML.");
+                return false;
+            }
+            free(any->value.mem);
+            any->value_type = LYD_ANYDATA_DATATREE;
+            any->value.tree = tree;
+        /* fallthrough */
+        case LYD_ANYDATA_DATATREE:
+            lyd_print_mem(&ptr, any->value.tree, LYD_XML, LYD_PRINT_WITHSIBLINGS);
+            break;
+        }
+
+        if (node->schema->nodetype == LYS_ANYXML) {
+            sr_val->type = HI_ANYXML_T;
+            sr_val->data.anyxml_val = ptr;
+        } else {
+            sr_val->type = HI_ANYDATA_T;
+            sr_val->data.anydata_val = ptr;
+        }
+        break;
+    default:
+        return false;
+    }
+
+    /* origin */
+    hiredis_edit_diff_get_origin(node, &origin, NULL);
+    sr_val->origin = origin;
+
+    return NULL;
+
+error:
+    free(sr_val->xpath);
+    return false;
+}
+
+static void * hiredis_realloc(void *ptr, size_t size)
+{
+    void *new_mem;
+
+    new_mem = realloc(ptr, size);
+    if (!new_mem) {
+        free(ptr);
+    }
+
+    return new_mem;
+}
+
+static bool hiredis_change_ly2sr(const struct lyd_node *node, const char *value_str,
+	const char *anchor, hi_val_t **sr_val_p)
+{
+    uint32_t end;
+    hi_val_t *sr_val;
+    struct lyd_node *node_dup = NULL;
+    const struct lyd_node *node_ptr;
+    LY_ERR lyrc;
+
+    sr_val = calloc(1, sizeof *sr_val);
+
+    if (value_str) {
+        /* replace the value in a node copy so that this specific one is stored */
+        assert(node->schema->nodetype & (LYS_LEAF | LYS_LEAFLIST));
+        lyrc = lyd_dup_single(node, NULL, 0, &node_dup);
+        if (lyrc) {
+            goto cleanup;
+        }
+
+        lyrc = lyd_change_term(node_dup, value_str);
+        if (lyrc && (lyrc != LY_EEXIST) && (lyrc != LY_ENOT)) {            
+            goto cleanup;
+        }
+        node_dup->parent = node->parent;
+        node_dup->flags |= node->flags & LYD_DEFAULT;
+
+        node_ptr = node_dup;
+    } else {
+        node_ptr = node;
+    }
+
+    /* fill the sr value */
+#if 0	
```

Comment:
please do not leave dead code in the code base.  If you want to keep it use a specific define that would turn it on.